magento_architecture.md

```diff
@@ -0,0 +1,36 @@
+# Magento Architecture & Diagrams
+
+
+## Relevant information
+
+* Doofinder module is ALWAYS installed via composer. 
+* Magento creates a store in Doofinder per website. It registers its website's id information inside its corresponding's store's options in order to get the correct products in the feed for the search engines corresponding to the store_views from that website.
+* If the Magento app is uninstalled then the associated Doofinder store is NOT deleted, neither its search engines nor indices.
+
+## Diagrams
+
+### Install Doofinder plugin from Magento
+
+```mermaid
+sequenceDiagram
+  Client ->> Magento: Install Doofinder app via composer
+  Magento ->> Client: truthy response
+  Client ->> Magento: bin/magento setup:upgrade (refreshes setup_module table)
+  Magento ->> Client: truthy response
+  Client ->> Magento: select platform's sector
+  Magento ->> Database: save sector
```

Comment:
Instead of Database I think Magento_Database would be a better name for clarification

Reply:
I agree